public/app/features/dashboard-scene/pages/DashboardScenePage.tsx

```diff
@@ -31,11 +30,22 @@ export function DashboardScenePage({ route, queryParams, location }: Props) {
   const { dashboard, isLoading, loadError } = stateManager.useState();
   // After scene migration is complete and we get rid of old dashboard we should refactor dashboardWatcher so this route reload is not need
   const routeReloadCounter = (location.state as any)?.routeReloadCounter;
+  const prevParams = useRef<Params<string>>(params);
```

Comment:
I had a feeling we would need useRef! to access prev params inside the useEffect without it re-triggering :)